dist/_hyperscript.js

```diff
@@ -5627,7 +5627,11 @@
                 }
             }
 
-            if (tokens.matchToken("index")) {
+            if (tokens.matchToken("index") || tokens.matchToken("by")) {
+                var identifierToken = tokens.requireTokenType("IDENTIFIER");
```

Comment:
dumb but can you move the "by" match inside the if statement to be symmetric w/ the other side?  Also should we add the "with" option as well?

```
  for x in foo indexed with ...
```

Reply:
no.

This if statement is

```
if(tokens.matchToken("index") || tokens.matchToken("by")) {
```

because it allows for

```
for x in foo by i
```

in addition to 

```
for x in foo indexed by i
```

If I move the "by" match inside the if statement that drastically changes what the PR does.

Instead of `for x in foo by i` it would instead allow

```
for x in foo index by i
```

in addition to 

```
for x in foo indexed by i
```

---

I am, however, ok with adding

```
for x in foo indexed with i
```

although I worry that at a certain point it starts running into a readability issue, especially if you also decide you want a with keyword that means something different somewhere else.

Reply:
mmm, I see

i don't like the `for x in foo by i` syntax, I think I like 2 minus the first option.

Are you good w/ that?

Reply:
yeah sure. Sounds good to me. I liked its succinctness but if you dont like it Im happy with people who care about it reading like english typing out `indexed by` and people who dont using `index`.

I will get to this soon, along with removing the build artifacts from the commit because im assuming you probably run the build yourself anyway for security reasons so that you KNOW whats in the gzip.

Im pretty sure I just need to remove the check for 'by' to achieve the desired syntax, and remove the test for it, about to go try it out

Reply:
done :)